deployment/common/changeset/example/link_transfer.go

```diff
@@ -0,0 +1,182 @@
+package example
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	owner_helpers "github.com/smartcontractkit/ccip-owner-contracts/pkg/gethwrappers"
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/changeset"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+)
+
+const MaxTimelockDelay = 24 * 7 * time.Hour
+
+type TransferConfig struct {
+	To    common.Address
+	Value *big.Int
+}
+
+type MCMSConfig struct {
+	ValidUntil   uint32        // unix time until the proposal will be valid
+	MinDelay     time.Duration // delay for timelock worker to execute the transfers.
+	OverrideRoot bool
+}
+
+type LinkTransferConfig struct {
+	Transfers  map[uint64][]TransferConfig
+	From       common.Address
+	McmsConfig *MCMSConfig
+}
+
+var _ deployment.ChangeSet[*LinkTransferConfig] = LinkTransfer
+
+// Validate checks that the LinkTransferConfig is valid.
+func (cfg LinkTransferConfig) Validate(e deployment.Environment) error {
+	ctx := e.GetContext()
+	// Check that Transfers map has at least one chainSel
+	if len(cfg.Transfers) == 0 {
+		return fmt.Errorf("transfers map must have at least one chainSel")
+	}
+
+	// Check transfers config values.
+	for chainSel, transfers := range cfg.Transfers {
+		_, err := chain_selectors.GetSelectorFamily(chainSel)
+		if err != nil {
+			return fmt.Errorf("invalid chain selector: %w", err)
+		}
+		chain := e.Chains[chainSel]
+		addrs, err := e.ExistingAddresses.AddressesForChain(chainSel)
+		if len(transfers) == 0 {
+			return fmt.Errorf("transfers for chainSel %d must have at least one LinkTransfer", chainSel)
+		}
+		totalAmount := big.NewInt(0)
+		linkState, err := changeset.MaybeLoadLinkTokenState(chain, addrs)
+		if err != nil {
+			return fmt.Errorf("error loading link token state during validation: %w", err)
+		}
+		for _, transfer := range transfers {
+			if transfer.To == (common.Address{}) {
+				return fmt.Errorf("'to' address for transfers  must be set")
+			}
+			if transfer.Value == nil {
+				return fmt.Errorf("value for transfers must be set")
+			}
+			if transfer.Value == big.NewInt(0) {
+				return fmt.Errorf("value for transfers must be non-zero")
+			}
+		}
+		// check that from address has enough funds for the transfers
+		balance, err := linkState.LinkToken.BalanceOf(&bind.CallOpts{Context: ctx}, cfg.From)
+		if balance.Cmp(totalAmount) < 0 {
+			return fmt.Errorf("timelock address does not have enough funds for transfers for chain selector %d", chainSel)
```

Comment:
`timelock address does not have enough` , it wont always be a timelock address right?

Reply:
yes this was renamed here https://github.com/smartcontractkit/chainlink/pull/15512/commits/03e93827fffe21bbf491f371b84918ae965ffa7e